recipes/args-parser/6.0.0.0/conanfile.py

```diff
@@ -0,0 +1,51 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ArgsParserConan(ConanFile):
+    name = "args-parser"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/igormironchik/args-parser"
+    license = "MIT"
+    description = "Small C++ header-only library for parsing command line arguments."
+    topics = ("conan", "args-parser", "argument", "parsing")
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.5",
+            "apple-clang": "10"
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+
+        compiler = str(self.settings.compiler)
+        if compiler not in self._compilers_minimum_version:
+            self.output.warn("Unknown compiler, assuming it supports at least C++14")
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < self._compilers_minimum_version[compiler]:
+            raise ConanInvalidConfiguration("args-parser requires a compiler that supports at least C++14")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "Args"), dst=os.path.join("include", "args-parser"))
+
+    def package_id(self):
+        del self.info.settings.compiler
+        self.info.header_only()
```

Comment:
The settings object is reset by calling `self.info.header_only`.

https://github.com/conan-io/conan/blob/f230bf104842c160be6ea377497151fe1ea64c38/conans/model/info.py#L575-L578


```suggestion
        self.info.header_only()
```